src/mixins/repetition.js

```diff
@@ -6,24 +6,54 @@ export const RepetitionMixin = (superclass) =>
         /**
          * Returns an array of coordinates (lattice points) to repeat the 3D objects bases on the number of repetitions.
          */
-        get repetitionCoordinates() {
+        repetitionCoordinates(value) {
             const basis = new Made.Basis({
                 ...this.basis.toJSON(),
                 elements: ["Si"],
                 coordinates: [[0, 0, 0]],
             });
-            const value = this.settings.repetitions;
             // avoid repeating in z direction if boundaries are enabled.
             const repetitions = [value, value, this.areNonPeriodicBoundariesPresent ? 1 : value];
             return Made.tools.basis.repeat(basis, repetitions).coordinates.map((c) => c.value);
         }
 
+        // eslint-disable-next-line class-methods-use-this
+        coordinatesByAxes(coordinates, repetitions) {
```

Comment:
What does this function do? Let's add a dosctring, explaining the parameter types as well